Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/RoutingAdapter.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzFramework/DocumentPropertyEditor/PropertyEditorNodes.h>
+#include <AzFramework/DocumentPropertyEditor/RoutingAdapter.h>
+
+namespace AZ::DocumentPropertyEditor
+{
+    void RoutingAdapter::ClearRoutes()
+    {
+        RemoveRoute(Dom::Path());
+    }
+
+    void RoutingAdapter::AddRoute(const Dom::Path& route, DocumentAdapterPtr adapter)
+    {
+        if (m_rootRouter != nullptr)
+        {
+            m_rootRouter->AddRoute(m_relativeRoutingPath / route, adapter);
+            return;
+        }
```

Comment:
This looks incorrect.
SetRouter accepts a path to route to the parent router on, but this doesn't actually check that path.

Furthermore, if the parent router is ever removed, it takes these routes with them.

Shouldn't the routes supplied to AddRoute always be added to this router adaptor.
So if a parent router adaptor is removed, then this router the user added for the particular adaptor still works.

Pretty much if the user which calls AddRoute doesn't know that a parent router adaptor was attached at the time, then it would be surprising if there Dom::Path route stopped working if some time later the parent router was removed.

Reply:
That path may not yet be valid to check at this time, but when `AdapterBuilder` support gets added for routers I intend to add more safety rails here.

As for the root router being destroyed, that's currently undefined behavior - nested adapters aren't intended to be set as the root for the purposes of a view, the root `RoutingAdapter` is always intended to be the source of truth, so if it's no longer around this adapter should no longer be valid - it's intended that these will get built in a top-down manner, and recreating the contents will actually refresh any nested adapters from the root down. Definitely some more work to be done on making this ergonomic though!

Reply:
In that case it seems the API is kind of backwards.
Wouldn't the parent router add children routers to it?